presto-main/src/main/java/com/facebook/presto/operator/JoinHash.java

```diff
@@ -71,6 +76,7 @@ public long getJoinPosition(int position, Page hashChannelsPage, Page allChannel
     @Override
     public long getJoinPosition(int position, Page hashChannelsPage, Page allChannelsPage, long rawHash)
     {
+        first = true;
```

Comment:
Yes, that's one of the reasons why I think decoupling `PositionLinks` from `JoinHash` is a way to go. I should have marked this as a `//TODO`